ledger/error/src/main/scala/com/daml/error/ErrorCode.scala

```diff
@@ -50,93 +52,74 @@ abstract class ErrorCode(val id: String, val category: ErrorCategory)(implicit
 
   implicit val code: ErrorCode = this
 
-  /** The error code string, uniquely identifiable by the error id, error category and correlation id.
+  /** The machine readable error code string, uniquely identifiable by the error id, error category and correlation id.
     * e.g. NO_DOMAINS_CONNECTED(2,ABC234)
     */
   def codeStr(correlationId: Option[String]): String =
     s"$id(${category.asInt},${correlationId.getOrElse("0").take(8)})"
 
-  /** Render the cause including codestring and error name */
+  /** @return message including error category id, error code id, correlation id and cause
+    */
   def toMsg(cause: => String, correlationId: Option[String]): String =
     s"${codeStr(correlationId)}: ${ErrorCode.truncateCause(cause)}"
 
-  def asGrpcStatus(err: BaseError)(implicit
-      loggingContext: ContextualizedErrorLogger
-  ): Status = {
-    val ErrorCode.StatusInfo(codeGrpc, message, contextMap, correlationId) =
-      getStatusInfo(err)
-
+  def asGrpcStatus(err: BaseError)(implicit loggingContext: ContextualizedErrorLogger): Status = {
+    val statusInfo = getStatusInfo(err)(loggingContext)
     // Provide error id and context via ErrorInfo
-    val maybeErrInfo =
-      if (!code.category.securitySensitive) {
-        val errInfoBld = com.google.rpc.ErrorInfo.newBuilder()
-        contextMap.foreach { case (k, v) => errInfoBld.putMetadata(k, v) }
-        errInfoBld.setReason(id)
-
-        // TODO error codes: Resolve dependency and use constant
-        //    val definiteAnswerKey = com.daml.ledger.grpc.GrpcStatuses.DefiniteAnswerKey
-        val definiteAnswerKey = "definite_answer"
-        err.definiteAnswerO.foreach { definiteAnswer =>
-          errInfoBld.putMetadata(definiteAnswerKey, definiteAnswer.toString)
-        }
-        Some(com.google.protobuf.Any.pack(errInfoBld.build()))
-      } else None
-
+    val errorInfo =
+      if (code.category.securitySensitive) {
+        None
+      } else {
+        val errorInfo = ErrorDetails.ErrorInfoDetail(
```

Comment:
Using our `ErrorDetails.ErrorInfoDetail` rather than `com.google.rpc.ErrorInfo.newBuilder`.
Similarly using `ErrorDetails.RetryInfoDetail`, `ErrorDetails.RequestInfoDetail` and `ErrorDetails.ResourceInfoDetail` below